docs/source/tools/speech_data_processor.rst

```diff
@@ -5,4 +5,134 @@ Speech Data Processor (SDP) is a toolkit to make it easy to:
+
+The ``DataEntry`` class is a dataclass which contains 2 attributes:
+
+1. ``data`` is an Optional dictionary containing items which represent the JSON manifest entry. ``data`` can also be ``None``. If a ``.process_dataset_entry(data_entry)`` method returns a ``DataEntry`` class where ``data is None``, then that utterance will be dropped from the output manifest.
+2. ``metrics``, which can be of any type, and are ``None`` by default. This variable is used by some variables to record summary statistics about the changes made to the dataset, these metrics are aggregated and can be displayed once every utterance has been processed by the processor.
+
+What happens in **BaseParallelProcessor.process()**?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We outline the ``.process()`` method of the ``BaseParallelProcessor`` class below:
+
+.. raw:: html
+
+    <div align="center">
+      <img src="https://mermaid.ink/img/pako:eNplUl1r6zAM_SvCFy4pbL3vvaVwu-59sL0tl6LESmqIP7DkjWzsv89O0rVjzosiHR8dHetdtV6T2qg-YjjB0-Fv7SAfTs2cqdWjUGAwDrYiuz0yPWDEYaDhIfqWmH1chzmqVts_GQOW5OR1rWaqcv4916pcZxq6jKaAkRb0tok7IBtkXO5BM4KmDtMgUIotOmgIEpMG8VOK1v0atH91g0cNEV9BoyBgEm9RTJvljbX6D7e3O9hfVOyvVURCfbToTEcs11pKocwbksC5PnWFyhB00VvIE7wYnxiWwY3rgbNNqwlnOpATRQLD4B2dhdxdhNx9t2PiOJYRmORITuJYlb85XEydFGDDErGVL4tn6gNcuA-Zm_GFwCf5McJvwL6P1KNQoYim5SlfTY7-At9BEmHQ0YdAenVucH_hv7_W3hmHg3mj40JWXYudX8lwGHD86rb4d7YtN6hd-Qo1Oa1ulKVo0ei8k-8lXatsps0ubnK47EVZrY8MLQ_-OLpWbSQmulEpZNvoYDDvrlWbDgemj0-10vX9" height=100% />
+    </div>
+
+
+**ModifyManifestTextProcessor**
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``ModifyManifestTextProcessor`` inherits from the ``BaseParallelProcessor`` class. 
+
+The ``ModifyManifestTextProcessor`` constructor takes in the following arguments: 
+* ``text_key`` (string) and ``pred_text_key`` (string): these parameters specify which keys in ``data_entry.data`` will be used for processing. (default: ``text_key="text"``, ``pred_text_key="pred_text"``, ie. by default the processor will refer to and modify the ``"text"`` and/or ``"pred_text"`` attributes of the input manifest).
+* ``test_cases`` (optional, list of dicts) - test cases for checking that the processor makes the changes that we are expecting.
+
+``ModifyManifestTextProcessor`` has the following methods: 
+* ``.test()``: this method makes sure that the output from the processor matches the expected output specified in the ``test_cases`` parameter.
+* ``.process_dataset_entry(data_entry)``: this method applies processing to a ``data_entry``. First, spaces are added to the start and end of the 'text' and 'pred_text' entries (if they exist), then the abstract method ``._process_dataset_entry(data_entry)`` is called. Then, any extra spaces (e.g. two spaces next to each other '  ') are removed from 'text' and 'pred_text' entries.
+* ``._process_dataset_entry(data_entry)``: this is an abstract method which will be over-written by children of ``ModifyManifestTextProcessor``.
```

Comment:
Hmm I see now what the . Is. Let's replace it with full base class name everywhere

Reply:
OK, I will put full base class name in some places, in others just remove the "."